include/uapi/sound/sof/tokens.h

```diff
@@ -90,6 +90,12 @@
 #define SOF_TKN_COMP_NUM_AUDIO_FORMATS		410
 #define SOF_TKN_COMP_NUM_SINK_PINS		411
 #define SOF_TKN_COMP_NUM_SOURCE_PINS		412
+/* The token for sink/source pin binding, it specifies the widget
+ * name that the sink/source pin is connected from/to.
+ */
+#define SOF_TKN_COMP_SINK_PIN_BINGING_WNAME	413
+#define SOF_TKN_COMP_SRC_PIN_BINGING_WNAME	414
+
```

Comment:
You need to add something in the commit message on the generic case, e.g.

An example of widget that has special pin is the smart
amplifier widget, its feedback pin has to be connected
to a capture DAI copier for codec feedback, while the
other pin has to be connected to a host DAI. The Pin ID for such special cases is defined in the topology, its value will be used during widget binding for correct route setup

Conversely the pin ID for 'generic' pins is not defined in the topology and allocated by the kernel. When only one pin is supported the pin ID shall always be zero. When more than one pin is used, the pin ID is determined with ida_alloc_range()